src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -205,20 +214,25 @@ class DeepDiveLogParser extends JavaTokenParsers {
     | "[" ~> cond <~ "]"
     )
 
-  def cqBodyAtom: Parser[Atom] =
-    relationName ~ "(" ~ repsep(expr, ",") ~ ")" ^^ {
-      case (r ~ "(" ~ patterns ~ ")") => Atom(r, patterns)
+  def atom = relationName ~ "(" ~ repsep(expr, ",") ~ ")" ^^ {
+    case (r ~ "(" ~ patterns ~ ")") => Atom(r, patterns)
+  }
+  def modifierAtom = (opt("NOT") ~ "EXIST" | "OUTER") ~ "[" ~ rep1sep(cqBody, ",") ~ "]" ^^ { case (m ~ _ ~ b ~ _) =>
```

Comment:
I'm a little worried about introducing negation without putting much thought into it, but that's probably fine since we don't plan to support recursion any time soon.

It may be slightly complicated since you also have to negate any Cond in the nested body (De Morgan's law).  Maybe it's better to keep this out from the parser and have a desugaring step, like the SemanticChecker because I think we want to print as it is when pretty printed.